### DISCUSSION TITLE
what's the difference between connected_components and vertex_components?

### DISCUSSION THREAD

**Discussion by chaohuang** [libigl/libigl#1918] [Q&A] [answered]
It seems that both `connected_components` and `vertex_components` return connected components based on vertices (i.e. connected if sharing vertices). What's the difference between them?

Thanks.

**Comment by alecjacobson** [comment, marked as answer]
I'm not sure why there're all those vertex_component overloads. The
relevant one is
template <typename DerivedF, typename DerivedC>
IGL_INLINE void vertex_components(
const Eigen::MatrixBase<DerivedF> & F,
Eigen::PlainObjectBase<DerivedC> & C); which takes the face list as input.
connected_component takes the adjacency matrix directly and acts on it as a
graph
